crates/relayer-cosmos/src/base/impls/chain.rs

```diff
@@ -192,4 +194,42 @@ where
 
         Ok(is_packet_received)
     }
+
+    async fn query_write_ack_event(
+        &self,
+        channel_id: &Self::ChannelId,
+        port_id: &Self::PortId,
+        counterparty_channel_id: &<CosmosChainWrapper<Counterparty> as OfaChainTypes>::ChannelId,
+        counterparty_port_id: &<CosmosChainWrapper<Counterparty> as OfaChainTypes>::PortId,
+        sequence: &<CosmosChainWrapper<Counterparty> as OfaChainTypes>::Sequence,
+    ) -> Result<Option<Self::WriteAcknowledgementEvent>, Self::Error> {
+        let status = self.query_chain_status().await?;
+        let chain_handle = self.chain.chain_handle();
+        let src_query_height =
+            Qualified::Equal(*CosmosChainWrapper::<Chain>::chain_status_height(&status));
+        let path_ident = PathIdentifiers {
+            port_id: port_id.clone(),
+            channel_id: channel_id.clone(),
+            counterparty_port_id: counterparty_port_id.clone(),
+            counterparty_channel_id: counterparty_channel_id.clone(),
+        };
+
+        let ibc_events =
+            query_write_ack_events(chain_handle, &path_ident, &[*sequence], src_query_height)
+                .map_err(Error::relayer)?;
+
+        let write_ack = ibc_events.into_iter().find_map(|event_with_height| {
+            let event = event_with_height.event;
+
+            if let IbcEvent::WriteAcknowledgement(write_ack) = event {
+                Some(write_ack)
+            } else {
+                None
+            }
+        });
+
+        // Would we ever get more than one WriteAcknowledgement?
```

Comment:
If the query is implemented correctly, there should only be one write acknowledgement. This is because there is only one sequence in the array passed to `query_write_ack_events`.